test/test_matmul_fp8_cpu.py

```diff
@@ -0,0 +1,87 @@
+# Owner(s): ["module: mkldnn"]
+
+from typing import Optional
+
+import torch
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+
+# Protects against includes accidentally setting the default dtype
+assert torch.get_default_dtype() is torch.float32
+
+e4m3_type = torch.float8_e4m3fn
+e5m2_type = torch.float8_e5m2
+E4M3_MAX_POS = torch.finfo(torch.float8_e4m3fn).max
+E5M2_MAX_POS = torch.finfo(torch.float8_e5m2).max
+
+
+class TestFP8MatmulCpu(TestCase):
+    def _test_tautological_mm(
+        self,
+        device: str = "cpu",
+        x_dtype: torch.dtype = e4m3_type,
+        y_dtype: torch.dtype = e4m3_type,
+        out_dtype: Optional[torch.dtype] = None,
+        size: int = 16,
+    ) -> None:
+        x_fp8 = torch.rand(size, size, device=device).to(x_dtype)
+        y_fp8 = torch.eye(size, device=device, dtype=y_dtype).t()
+        out_fp32 = torch.mm(x_fp8.to(torch.float), y_fp8.to(torch.float))
+        scale_a = torch.tensor(1.0, device=device)
+        scale_b = torch.tensor(1.0, device=device)
+        out_fp8 = torch._scaled_mm(x_fp8, y_fp8, scale_a, scale_b, out_dtype=out_dtype)
+        if out_dtype is not None:
+            self.assertEqual(out_dtype, out_fp8.dtype)
+        self.assertEqual(out_fp32, out_fp8.to(torch.float))
+
+    def test_float8_basics(self, device) -> None:
```

Comment:
It should be coniditional on `skipIfNoMLKDNN`, shoudn't it?

Reply:
For CPU, we have a fallback function `_scaled_mm_out_cpu_emulated`.